src/trace/context/extractors/sqs.ts

```diff
@@ -3,18 +3,29 @@ import { EventTraceExtractor } from "../extractor";
 import { TracerWrapper } from "../../tracer-wrapper";
 import { logDebug } from "../../../utils";
 import { SpanContextWrapper } from "../../span-context-wrapper";
+import { XrayService } from "../../xray-service";
 
 export class SQSEventTraceExtractor implements EventTraceExtractor {
   constructor(private tracerWrapper: TracerWrapper) {}
 
   extract(event: SQSEvent): SpanContextWrapper | null {
-    const headers = event?.Records?.[0]?.messageAttributes?._datadog?.stringValue;
-    if (headers === undefined) return null;
-
     try {
-      const traceContext = this.tracerWrapper.extract(JSON.parse(headers));
-      if (traceContext === null) return null;
+      let preparedHeaders;
+      const headers = event?.Records?.[0]?.messageAttributes?._datadog?.stringValue;
+      if (headers !== undefined) {
+        preparedHeaders = JSON.parse(headers);
```

Comment:
nit, mostly because we are parsing in both? wdyt?
```suggestion
        parsedHeaders = JSON.parse(headers);
```